python/plugins/processing/gui/ExtentSelectionPanel.py

```diff
@@ -205,10 +152,9 @@ def setValueFromRect(self, r):
         self.dialog.activateWindow()
 
     def getValue(self):
-        if unicode(self.leText.text()).strip() != '':
-            return unicode(self.leText.text())
-        else:
-            return self.getMinCoveringExtent()
+        if unicode(self.leText.text()).strip() == '':
+            return None
```

Comment:
I think that we should not set ParameterExtent optional by default, as we don't know if there if a ParameterDataObject on extent parameter creation.

If we have a pointer to algorithm (like with the QObject inheritance with parent/child relationship), we could test in ParameterExtent.setValue if we can use auto extent.

Another solution should be to set correct optional value in all algorithms.